server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1424,7 +1424,7 @@ private SearchSourceBuilder parseXContent(XContentParser parser, boolean checkTr
                             sorts = new ArrayList<>(SortBuilder.fromXContent(parser));
                         } else if (RESCORE_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
                             rescoreBuilders = new ArrayList<>();
-                            rescoreBuilders.add(RescorerBuilder.parseFromXContent(parser));
+                            rescoreBuilders.add(RescorerBuilder.parseFromXContent(parser, searchUsage::trackQueryUsage));
```

Comment:
🤔 

This might be ok, but one key issue I can think of is that NamedXContent is only unique per parent class. Meaning we could have name conflicts if a query of the same name.

The other option is adding a whole new section `searchUsage::trackRescorerUsage`. This seems like the most correct thing, but I know adding a new section and then actually seeing it in telemetry, etc. is a huge pain.

@jimczi ^ what say you?